### ISSUE TITLE
Pass dynamic variables when calling definitions

### ISSUE THREAD

**Issue by aschwin** [zircote/swagger-php#398]
Bob, would it be possible to pass variables when calling definitions? Like, when you have a sort of template where only one part of the final definition name is different, it would be welcome to be able to call the definition with a variable/parameter.

Not to confuse it with previous issue #229, which is similar but not the same. 

Like in the examples beneath, the only difference should be CustomerList in CustomerPaginated and Address in AddressPaginated. But this is not possible in the current setup I guess?

This is just one example and our application is littered with this repeated pattern for each paginated result.

If you could help us out, that would be very welcome. Thanks.

 ```
     * @SWG\Definition(
     *   definition="CustomerPaginatedResponse",
     *   type="object",
     *   allOf={
     *     @SWG\Schema(ref="#/definitions/Pagination"),
     *     @SWG\Schema(ref="#/definitions/CustomerEmbedded")
     *   }
     * )
     *
     * @SWG\Definition(
     *    definition="CustomerEmbedded",
     *    @SWG\Property(property="_embedded", type="array", @SWG\Items(ref="#/definitions/CustomerItems"))
     * )
     *
     * @SWG\Definition(
     *    definition="CustomerItems",
     *    @SWG\Property(property="items", type="array", @SWG\Items(ref="#/definitions/CustomerPaginated"))
     * )
     *
     * @SWG\Definition(
     *    definition="CustomerPaginated",
     *    type="object",
     *    allOf={
     *      @SWG\Schema(ref="#/definitions/CustomerList"),
     *      @SWG\Schema(ref="#/definitions/SelfLink")
     *   }
     * )
```

```
     * @SWG\Definition(
     *   definition="AddressPaginatedResponse",
     *   type="object",
     *   allOf={
     *     @SWG\Schema(ref="#/definitions/Pagination"),
     *     @SWG\Schema(ref="#/definitions/AddressEmbedded")
     *   }
     * )
     *
     * @SWG\Definition(
     *    definition="AddressEmbedded",
     *    @SWG\Property(property="_embedded", type="array", @SWG\Items(ref="#/definitions/AddressItems"))
     * )
     *
     * @SWG\Definition(
     *    definition="AddressItems",
     *    @SWG\Property(property="items", type="array", @SWG\Items(ref="#/definitions/AddressPaginated"))
     * )
     *
     * @SWG\Definition(
     *    definition="AddressPaginated",
     *    type="object",
     *    allOf={
     *      @SWG\Schema(ref="#/definitions/Address"),
     *      @SWG\Schema(ref="#/definitions/SelfLink")
     *   }
     * )
```

**Comment by bfanger** [comment]
_Nice to know a former colleague is also using swagger-php_

It's not possible to pass variables. However it should be possible to remove lots of duplication by creating your own annotations and generate regular swagger annotations by building a processor.

For example `@MyCompany\PaginatedResponse(items="#/definitions/Adress")`

swagger-php also doesn't offer any convenience shortcuts for [HAL](https://tools.ietf.org/html/draft-kelly-json-hal-08)
Building a set of helper annotations like `@HAL\Embedded()` would be beneficial to you and others #85

Sadly, although swagger-php allows third-party processors and annotations, it isn't documented and I don't know of any examples. 
Let me know if you encounter any issues.

**Comment by aschwin** [comment]
Yes, Swagger is in full swing for our applications. And the main language is currently still PHP.

I think we then first have to know more about annotations within PHP in general and then build from there how swagger-php is using this and how custom annotations and processors should be implemented. Especially if we want to leverage HAL annotations.

For now, we manage to have the needed and desired output, so it's not in a rush.

Thanks for the reply. I'll post an update if we have questions or if we start building our own processors/annotations.
